ext/standard/dns.c

```diff
@@ -210,10 +211,10 @@ PHP_FUNCTION(gethostbyname)
 		Z_PARAM_STRING(hostname, hostname_len)
 	ZEND_PARSE_PARAMETERS_END();
 
-	if(hostname_len > MAXFQDNLEN) {
+	if (hostname_len > MAXFQDNLEN) {
 		/* name too long, protect from CVE-2015-0235 */
-		php_error_docref(NULL, E_WARNING, "Host name is too long, the limit is %d characters", MAXFQDNLEN);
-		RETURN_STRINGL(hostname, hostname_len);
+		zend_argument_value_error(1, "must be less than %d characters", MAXFQDNLEN);
```

Comment:
what about `cannot be longer than %d characters`? As far as I see the condition, `less than MAXFQDNLEN` isn't true, because being equal is also okay.